django/core/exceptions.py

```diff
@@ -192,3 +192,8 @@ def __repr__(self):
 class EmptyResultSet(Exception):
     """A database query predicate is impossible."""
     pass
+
+
+class NoSuchShell(Exception):
```

Comment:
Agreed -- putting that in `shell.py` should be fine. Maybe that's better to inherit from `ImportError` and call it `ShellImportError`?